### PR TITLE
Added -nkey flag to tools for nkey authentication support

### PR DESCRIPTION
**Pull request by aricart**

### CODE REVIEW

File: tools/nats-bench/nats-bench.go [hunk 1]
```diff
@@ -80,15 +99,27 @@ func main() {
...
+
+		go runSubscriber(nc, &startwg, &donewg, *numMsgs, *msgSize)
 	}
 	startwg.Wait()
 
 	// Now Publishers
 	startwg.Add(*numPubs)
 	pubCounts := bench.MsgsPerClient(*numMsgs, *numPubs)
 	for i := 0; i < *numPubs; i++ {
-		go runPublisher(&startwg, &donewg, opts, pubCounts[i], *msgSize)
```
Comment (derekcollison) [inline tools/nats-bench/nats-bench.go]: Why did you remove helpers here?

File: tools/nats-echo/nats-echo.go [hunk 2]
```diff
@@ -65,11 +66,24 @@ func main() {
...
+
 	// Use UserCredentials
 	if *userCreds != "" {
 		opts = append(opts, nats.UserCredentials(*userCreds))
 	}
 
+	// Use Nkey authentication.
+	if *nkeyFile != "" {
+		opt, err := nats.NkeyOptionFromSeed(*nkeyFile)
+		if err != nil {
```
Comment (derekcollison) [inline tools/nats-echo/nats-echo.go]: again ignore err in all examples and let Connect pick it up?

File: tools/nats-bench/nats-bench.go (hunk 1 above)
Comment (aricart) [reply tools/nats-bench/nats-bench.go]: go-nats-example/nats-bench.go was out of sync with go-nats version, the current version is a copy from go-nats. Helper is still there

File: tools/nats-echo/nats-echo.go (hunk 2 above)
Comment (aricart) [reply tools/nats-echo/nats-echo.go]: same comment - I think the examples should show error handling, if what we think we are saying is that adding an option shouldn't return an error until a connect, we need to address that instead

Comment (aricart) [reply tools/nats-echo/nats-echo.go]: +derekcollison Added goreleaser so the release page has pre-built binaries like we do for nsc, ngs, etc. The current release binaries date back to june 6, 2017!

Comment (derekcollison) [reply tools/nats-echo/nats-echo.go]: I believe Connect will return the error, so you don't have to process in place and can make the example a little shorter. Be good to try out though..

Comment (aricart) [reply tools/nats-echo/nats-echo.go]: Without the error, you get the error: `nats: Authorization Violation`
With the error checking you get an error that is much better `nats: open /tmp/foo: no such file or directory`

Comment (aricart) [reply tools/nats-echo/nats-echo.go]: Since this is tooling and not a code snippet I think the targetted error message is much better.

Comment (derekcollison) [reply tools/nats-echo/nats-echo.go]: ok, wonder if we should return error for creds option as well.

**Comment by derekcollison** [comment]
LGTM
